R/exclude-potential-pcos.R

```diff
@@ -0,0 +1,54 @@
+#' Exclude metformin purchases potentially for the treatment of polycystic ovary syndrome.
+#'
+#' Takes the output from `include_gld_purchases()` and `bef` (information on sex and date
+#' of birth) to do the exclusions.
+#' This function only performs a filtering operation so outputs the same structure and
+#' variables as the input from `include_gld_purchases()`, except the addition of a logical
+#' helper variable `no_pcos` that is used in later functions.
+#' After these exclusions are made, the output is used by `exclude_pregnancy()`.
+#'
+#' @param gld_purchases The output from `include_gld_purchases()`.
+#' @param bef The `bef` register.
+#'
+#' @return The same type as the input data, default as a [tibble::tibble()].
+#'    It also has the same columns as `include_gld_purchases()`, except for a logical
+#'    helper variable `no_pcos` that is used in later functions.
+#' @keywords internal
+#' @inherit algorithm seealso
+#'
+#' @examples
+#' exclude_potential_pcos(
+#'   gld_purchases = include_gld_purchases(register_data$lmdb),
+#'   bef = register_data$bef
+#' )
```

Comment:
```suggestion
#' @examples
#' \dontrun{
#' exclude_potential_pcos(
#'   gld_purchases = include_gld_purchases(register_data$lmdb),
#'   bef = register_data$bef
#' )
#' }
```

Since this is an internal function, we tell R not to run this example code when doing automated checks. This is for use to use to test out.